tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,247 @@
+<?php
+/**
+ * This script is written for a one time use only to normalise the consent feature of candidate parameters. 
+ * 
+ * New normalised tables will be populated with existing data from `participant_status` and `consent_info_history`.
+ * 
+ * A SQL file is exported to path: loris_root/SQL/Cleanup_patches/delete_old_consent_tables.sql
+ * after running this script. Source this file to delete the now deprecated tables and columns.
+ * 
+ * affected tables:
+ * - participant_status
+ * - consent_info_history
+ * - consent
+ * - candidate_consent_rel
+ * - candidate_consent_history
+ * 
+ * PHP Version 7
+ * 
+ * @category Main 
+ * @package  Loris
+ * @author   Zaliqa Rosli <zaliqa.rosli@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris
+ */
+
+require_once 'generic_includes.php';
+
+$config = NDB_Config::singleton();
+$db     =& Database::singleton();
+
+// Get consent details from Config.xml
+$consentConfig = $config->getSettingFromXML('ConsentModule');
+$useConsent = $consentConfig['useConsent'];
+$consents = $consentConfig['Consent'];
+
+///////////////////////////////////////
+// VALIDATE EXISTING CONSENT DETAILS //
+///////////////////////////////////////
+
+$errors = [];
+// Check for when consent columns exist in participant_status but not in Config.xml
+$columnQuery = 'SELECT Column_name FROM Information_schema.columns WHERE
+                Column_name LIKE "%consent%" AND Table_name LIKE "participant_status"';
```

Comment:
do all consent types columns in the config always have "consent" in their name?

Reply:
from what i've seen on CCNA and IBIS, yes. it's the only naming "standard" i have found